theforeman.org/scripts/debian/stage_packages.sh

```diff
@@ -7,19 +7,46 @@ set -xe
 # * the web node must have the freight class applied
 
 # Find the deps in the build-dir from the previous step
-DEB_PATH=./debian/${os}/build-${project}
+for possible_path in ./debian/${os}/build-${project} ./plugins/build-${project} ./dependencies/${os}/build-${project}; do
+  if [[ -d "${possible_path}" ]]; then
+    DEB_PATH="${possible_path}"
+  fi
+done
```

Comment:
Just for safety, could we add some code after this loop? Something like 
```suggestion
for possible_path in ./debian/${os}/build-${project} ./plugins/build-${project} ./dependencies/${os}/build-${project}; do
  if [[ -d "${possible_path}" ]]; then
    DEB_PATH="${possible_path}"
  fi
done

if [[ -z $DEB_PATH ]] ; then
  echo "No build-dir path found"
  exit 1
fi
```

I also wonder about the chance where it found something that wasn't intended, but I think that chance is small since projects only live in one directory and I don't recall us ever moving something. Famous last words, but I think it's safe to ignore that case for now.

Reply:
There is a `set -e` at the top of this script, so I'd expect it to abort if it tries to access an empty `DEB_PATH`, so the safeguard is already there?

Reply:
Should we just `set -u`?

Reply:
Could be, but I must admit I don't have a lot of experience with that. If it'd happen more often a clear error message can be helpful but I think we'd be fine with it given the chance is small.